Tavisca.Bootcamp.LanguageBasics.Exercise2/ForumPostEasy.cs

```diff
@@ -0,0 +1,96 @@
+            }
+            /*boundary condition to check if the given time is 23:59:59 few seconds ago
+             * then our minTimeSpan will be 23:59:59 and maxTimeSpan will be 00:00:58 then
+             * time 00:00:00 is lexicographically smaller
+             */
+            if(TimeSpan.Compare(maxTimeSpan.Min(), minTimeSpan.Max())==-1)
+            {
+                if (TimeSpan.Compare(minTimeSpan.Max(), new TimeSpan(0, 0, 0)) == -1 && (TimeSpan.Compare(maxTimeSpan.Min(), new TimeSpan(0, 0, 0)) == 1 || TimeSpan.Compare(maxTimeSpan.Min(), new TimeSpan(0, 0, 0)) == 0))
+                {
+                    return "00:00:00";
+                }
+                else
+                {
+                    return "impossible";
+                }
+            }
+            else
+            {
+                return minTimeSpan.Max().ToString();
+            }
+        }
+
+        private static void CalculateMinAndMaxTimePossible(TimeSpan[] minTimeSpan, TimeSpan[] maxTimeSpan, int i, string[] exactPostTime, string[] showPostTime)
+        {
+            String[] splitGivenTime = exactPostTime[i].Split(':');
+            TimeSpan temp = new TimeSpan(Int32.Parse(splitGivenTime[0]), Int32.Parse(splitGivenTime[1]), Int32.Parse(splitGivenTime[2]));
+            if (showPostTime[i].Contains("second"))
+            {
+                minTimeSpan[i] = temp;
+                maxTimeSpan[i] = new TimeSpan(0, 0, 59) + minTimeSpan[i];
+                minTimeSpan[i] = RemoveDaysIfTimeExceedsTwetyFourHours(minTimeSpan[i]);
+                maxTimeSpan[i] = RemoveDaysIfTimeExceedsTwetyFourHours(maxTimeSpan[i]);
+            }
+            if (showPostTime[i].Contains("minute"))
+            {
+                minTimeSpan[i] = new TimeSpan(0, Int32.Parse(showPostTime[i].Split(' ')[0]), 0) + temp;
+                maxTimeSpan[i] = new TimeSpan(0, 0, 59) + minTimeSpan[i];
+                minTimeSpan[i] = RemoveDaysIfTimeExceedsTwetyFourHours(minTimeSpan[i]);
+                minTimeSpan[i] = RemoveDaysIfTimeExceedsTwetyFourHours(minTimeSpan[i]);
+                maxTimeSpan[i] = RemoveDaysIfTimeExceedsTwetyFourHours(maxTimeSpan[i]);
+            }
+            if (showPostTime[i].Contains("hour"))
+            {
+                minTimeSpan[i] = new TimeSpan(Int32.Parse(showPostTime[i].Split(' ')[0]), 0, 0) + temp;
+                maxTimeSpan[i] = new TimeSpan(0, 59, 59) + minTimeSpan[i];
+                minTimeSpan[i] = RemoveDaysIfTimeExceedsTwetyFourHours(minTimeSpan[i]);
+                maxTimeSpan[i] = RemoveDaysIfTimeExceedsTwetyFourHours(maxTimeSpan[i]);
+            }
+        }
+        /* when time is 23:59:59 few seconds ago then adding
+         * 59 seconds will give 1.00:00:58 where 1 denotes
+         * days so removing the days from the time
+        */
+        private static TimeSpan RemoveDaysIfTimeExceedsTwetyFourHours(TimeSpan timeSpan)
```

Comment:
If you use dateTime, you don't need to do such manipulation which can save a lot of effort. Given that you can use the date factor from the type to avoid the day level calculations.
You can directly compare the dates after sorting them out.